plugin/src/main/java/org/opensearch/ml/action/model_group/TransportUpdateModelGroupAction.java

```diff
@@ -146,12 +156,42 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<MLUpda
                                         mlModelGroup.getTenantId(),
                                         wrappedListener
                                     )) {
-                                    if (modelAccessControlHelper.isSecurityEnabledAndModelAccessControlEnabled(user)) {
-                                        validateRequestForAccessControl(updateModelGroupInput, user, mlModelGroup);
+                                    // TODO: Remove this feature flag check once feature is GA, as it will be enabled by default
+                                    if (isResourceSharingFeatureEnabled) {
+                                        ResourceSharingClient resourceSharingClient = ResourceSharingClientAccessor
+                                            .getInstance()
+                                            .getResourceSharingClient();
+                                        resourceSharingClient
+                                            .verifyResourceAccess(modelGroupId, ML_MODEL_GROUP_INDEX, ActionListener.wrap(isAuthorized -> {
+                                                if (!isAuthorized) {
+                                                    listener
+                                                        .onFailure(
+                                                            new OpenSearchStatusException(
+                                                                "User "
+                                                                    + user.getName()
+                                                                    + " is not authorized to update ml-model-group: "
+                                                                    + mlModelGroup.getName(),
+                                                                RestStatus.FORBIDDEN
+                                                            )
+                                                        );
+                                                    return;
+                                                }
+                                                // For backwards compatibility we still allow storing backend_roles data in ml_model_group
+                                                // index
+                                                updateModelGroup(modelGroupId, r.source(), updateModelGroupInput, wrappedListener, user);
```

Comment:
So if user wants to update resource sharing fields through update model group API, how do we allow it?

Reply:
Users will now have to explicitly "share" the resource. Security has exposed "share" and "revoke" java APIs which can be consumed by ML in their respective handlers to update model group access.